core/deployment/src/main/java/io/quarkus/deployment/dev/IsolatedDevModeMain.java

```diff
@@ -334,12 +334,18 @@ public void close() {
         fsWatchUtil.shutdown();
         try {
             stop();
+            if (RuntimeUpdatesProcessor.INSTANCE == null) {
+                throw new IllegalStateException(
+                        "Hot deployment of the application is not supported when updating the quarkus version. The application needs to be stopped and dev-mode started up again");
```

Comment:
```suggestion
                        "Hot deployment of the application is not supported when updating the Quarkus version. The application needs to be stopped and dev mode started up again");
```
(we call it dev mode without dash in the guides if I'm not mistaken)

Reply:
Good point